### DISCUSSION TITLE
Accumulating many hidden terminal instances on mac

### DISCUSSION THREAD

**Discussion by rcorre** [kovidgoyal/kitty#7510] [Q&A] [answered]
Hi, I'm not sure if this is a Kitty problem, but I haven't seen it with other terminals. I'm on a mac, using `skhd` to launch kitty:

```
cmd - return : kitty -d ~
```

Over time, I tend to accumulate tons of kitty instances. They are hidden, and I can't find them on any workspace, but they show up in the cmd+tab  switcher:

![image](https://github.com/kovidgoyal/kitty/assets/2496231/6a0e404f-f97a-42d3-9589-7bc883203ea1)

If I cmd+tab to select one of them, I don't see a terminal. I don't have any options like "bring to front"

![image](https://github.com/kovidgoyal/kitty/assets/2496231/61b30418-f3a9-43ff-bc66-dc010998e01e)

Does anyone have a clue what might cause this? My current solution is to periodically `pkill` all `kitty` instances, which also kills terminals I care about.

kitty.conf:

```
font_family Hack
font_size 13.0
cursor_shape block
cursor_blink_interval -1
enable_audio_bell no
hide_window_decorations titlebar-and-corners
window_margin_width 2.0
macos_option_as_alt both
shell_integration no-cursor
```

`kitty 0.34.1`

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.macos_quit_when_last_window_closed

**Reply by rcorre** [reply]
That was it, thank you!
